Build failure with ocaml 4.07.0

```
File "zeroinstall/command.ml", line 7, characters 0-19:
Warning 44: this open statement shadows the module identifier String (which is later used)
File "zeroinstall/command.ml", line 50, characters 43-63:
Error: Type [ `For_each ] Element.t is not a subtype of
         arg_parent = [ `Command | `For_each | `Runner ] Element.t
       The first variant type does not allow tag(s) `Command, `Runner
Command exited with code 2.
make[2]: *** [ocaml] Error 10
make[1]: *** [all] Error 2
make: *** [all] Error 2
```

Running macOS 10.14 with Opam 1.2.2, OCaml 4.07.0.

Running a patched `ocaml` does indeed seem to resolve the issue, thanks.

On the flip-side, hello race condition, heh:

```
[ -d "/private/tmp/zero-install-20180810-83756-1h0gcw7/0install-2.13/build" ] || mkdir "/private/tmp/zero-install-20180810-83756-1h0gcw7/0install-2.13/build"
[ -d "/private/tmp/zero-install-20180810-83756-1h0gcw7/0install-2.13/build" ] || mkdir "/private/tmp/zero-install-20180810-83756-1h0gcw7/0install-2.13/build"
mkdir: /private/tmp/zero-install-20180810-83756-1h0gcw7/0install-2.13/build: File exists
[ -d "/private/tmp/zero-install-20180810-83756-1h0gcw7/0install-2.13/dist" ] || mkdir "/private/tmp/zero-install-20180810-83756-1h0gcw7/0install-2.13/dist"
make: *** [all] Error 1
make: *** Waiting for unfinished jobs....
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -10,6 +10,7 @@ env:
   matrix:
     - OCAML_VERSION=4.03
     - OCAML_VERSION=4.06
+    - OCAML_VERSION=4.07
 os:
   - linux
   - osx

--- a/ocaml/support/common.ml
+++ b/ocaml/support/common.ml
@@ -39,13 +39,6 @@ module StringMap = struct
 end
 module StringSet = Set.Make(String)
 
-module String = struct
-  include String
-  let capitalize_ascii x = String.capitalize x
-  let uppercase_ascii x = String.uppercase x
-  let lowercase_ascii x = String.lowercase x
-end [@@ocaml.warning "-3"]
-
 (** Define an interface for interacting with the system, so we can replace it
     in unit-tests. *)
 class type filesystem =

--- a/ocaml/zeroinstall/command.ml
+++ b/ocaml/zeroinstall/command.ml
@@ -28,6 +28,9 @@ let expand_arg arg env =
 
 type arg_parent = [`Command | `Runner | `For_each] Element.t
 
+(* This is a work-around for https://caml.inria.fr/mantis/view.php?id=7824 in OCaml 4.07.0 *)
+let as_arg_parent x = (x :> arg_parent)
+
 (* Return a list of string arguments by expanding <arg> and <for-each> children of [elem] *)
 let get_args elem env =
   let rec get_args_loop elem =
@@ -47,11 +50,11 @@ let get_args elem env =
           | x::xs ->
               let old = Env.get env "item" in
               Env.put env "item" x;
-              let new_args = get_args_loop (node :> arg_parent) in
+              let new_args = get_args_loop (as_arg_parent node) in
               old |> if_some (Env.put env "item");
               new_args @ (loop xs) in
         loop (Str.split_delim (Str.regexp_string separator) source)
-  in get_args_loop (elem :> arg_parent)
+  in get_args_loop (as_arg_parent elem)
 
 let find_ex role impls =
   Selections.RoleMap.find role impls

